### ISSUE TITLE
API: Get all workspaces of user returns every workspace as default

### ISSUE THREAD

**Issue by muntaxir4** [keyshade-xyz/keyshade#587]
**Describe the bug**
All returned workspaces from `/workspace` contains `isDefault: true`, whereas only one workspace should be a default one. This happens when a user becomes a member of host's default workpsace.

**Endpoints Affected**
- `/workspace`
- `/workspace/:workspaceSlug`

**To Reproduce**
1. User1 invites User2 to their default workspace.
2. User2 requests to view all of their accessible workspaces from `/workpsace`. Each of the workspace has `isDefault` set to `true`.

**Expected behavior**
A single Default workspace of User2 should have isDefault:true.

**Comment by zaheer-Khan3260** [comment]
Hey @rajdip-b, Please assign it to me.

**Comment by muntaxir4** [comment]
> Hey @rajdip-b, Please assign it to me.

You can take up this issue using `/attempt`

**Comment by csehatt741** [comment]
@zaheer-Khan3260, @muntaxir4, any progress here?

I can take a look at this issue, just assign it to me, please.

**Comment by rajdip-b** [comment]
Sure you ca go for it.

**Comment by csehatt741** [comment]
I`ve managed to reproduce this issue locally.

Do you mean only the default workspace of User2 should have `isDefault: true`, while the default workspace of User1 should have `isDefault: false`?

**Comment by rajdip-b** [comment]
Yes. Every user has a default workspace. If you get invited to someone else's default workspace, you still own a default workspace of your own.

The issue is, right now we are just fetching every workspace without processing the `ownerId` field.

So, what we want is, before sending out the workspace, we want to do a pre-processing that checks the following clause:
```javascript
workspace.isDefault = workspace.isDefault && workspace.ownerId == user.id
```
